Classes/Issues/Issue+IssueType.swift

```diff
@@ -93,32 +93,56 @@ extension IssueOrPullRequestQuery.Data.Repository.IssueOrPullRequest.AsIssue: Is
-                    title: unlabeled.label.name,
-                    color: unlabeled.label.color,
-                    date: date,
-                    type: .removed,
-                    repoOwner: owner,
-                    repoName: repo,
-                    width: width
-                )
-                results.append(model)
+                    let newDetails = IssueLabeledModel.LabelDetails(
+                        id: unlabeled.fragments.nodeFields.id,
+                        title: unlabeled.label.name,
+                        color: unlabeled.label.color,
+                        date: date,
+                        type: .removed)
+                    var removed = [IssueLabeledModel.LabelDetails]()
+                    var added = [IssueLabeledModel.LabelDetails]()
+
+                    if let lastModel = results.last as? IssueLabeledModel,
+                        lastModel.actor == unlabeled.actor?.login {
+                        added = lastModel.added
+                        removed = lastModel.removed
+                        removed.append(newDetails)
+                        results.removeLast()
+                    }
+                    let newModel = IssueLabeledModel(
+                        actor: unlabeled.actor?.login ?? Constants.Strings.unknown,
+                        added: added,
+                        removed: removed,
+                        repoOwner: owner,
+                        repoName: repo,
+                        width: width)
+                    results.append(newModel)
             } else if let labeled = node.asLabeledEvent,
                 let date = labeled.createdAt.githubDate {
-                let model = IssueLabeledModel(
-                    id: labeled.fragments.nodeFields.id,
-                    actor: labeled.actor?.login ?? Constants.Strings.unknown,
-                    title: labeled.label.name,
-                    color: labeled.label.color,
-                    date: date,
-                    type: .added,
-                    repoOwner: owner,
-                    repoName: repo,
-                    width: width
-                )
-                results.append(model)
+                    let newDetails = IssueLabeledModel.LabelDetails(
+                        id: labeled.fragments.nodeFields.id,
+                        title: labeled.label.name,
+                        color: labeled.label.color,
+                        date: date,
+                        type: .added)
+                    var removed = [IssueLabeledModel.LabelDetails]()
+                    var added = [IssueLabeledModel.LabelDetails]()
+
+                    if let lastModel = results.last as? IssueLabeledModel,
+                        lastModel.actor == labeled.actor?.login {
+                        added = lastModel.added
+                        removed = lastModel.removed
+                        removed.append(newDetails)
```

Comment:
`newDetails` only gets appended in this block, which is why it breaks for events with a single label